src/libraries/System.ComponentModel.TypeConverter/src/System/ComponentModel/PropertyDescriptor.cs

```diff
@@ -183,6 +184,7 @@ public override bool Equals(object obj)
         /// <summary>
         /// Creates an instance of the specified type.
         /// </summary>
+        [UnsupportedOSPlatform("browser")]
```

Comment:
I'm wondering if we mis-annotated `TypeDescriptor.CreateInstance` as not supported on browser in [https://github.com/dotnet/runtime/pull/41094](https://github.com/dotnet/runtime/pull/41094).  By default, it is just using reflection, so I'm not understanding what about it is unsupported.

@mdh1418 @steveisok @safern

Reply:
It was included in the unsupported APIs list: https://github.com/dotnet/designs/blob/main/accepted/2020/blazor-unsupported-apis/blazor-unsupported-apis.md#systemcomponentmodel
![image](https://user-images.githubusercontent.com/45579687/99576703-30293080-298f-11eb-8422-501ae4a0f7b9.png)
I guess it throws PNSE somewhere as the list is produced by `We use the scanner (https://github.com/dotnet/platform-compat) to gather the list of all the APIs throwing PNSE on Browser WASM at the moment`

Reply:
I just tried some of these methods in a 5.0 Blazor WASM app, and they work just fine.

In the `FetchData.razor` file of a default `dotnet new blazorwasm` app, change the code at the bottom to:

```C#
@code {
    private WeatherForecast[] forecasts;

    protected override async Task OnInitializedAsync()
    {
        //forecasts = await Http.GetFromJsonAsync<WeatherForecast[]>("sample-data/weather.json");

        var d = System.ComponentModel.TypeDescriptor.GetProvider(typeof(MyClass));
        MyClass c = (MyClass)d.CreateInstance(null, typeof(MyClass), null, null);
        c.Child = new MyChild();

        var p = d.GetTypeDescriptor(c).GetProperties().Find("Child", ignoreCase: true);
        var converter = p.Converter;

        forecasts = new[] {
            new WeatherForecast() { Summary = "This is a converter: " + converter.ToString() }
        };
    }

    public class WeatherForecast
    {
        public DateTime Date { get; set; }

        public int TemperatureC { get; set; }

        public string Summary { get; set; }

        public int TemperatureF => 32 + (int)(TemperatureC / 0.5556);
    }

    class MyClass
    {
        public string Name { get; set; }
        [System.ComponentModel.TypeConverter(typeof(MyChildConverter))]
        public MyChild Child { get; set; }
    }

    class MyChild
    {
        public string Name { get; set; }
    }

    class MyChildConverter : System.ComponentModel.TypeConverter
    {
        public override bool CanConvertTo(System.ComponentModel.ITypeDescriptorContext context, Type destinationType)
        {
            if (destinationType == typeof(string))
            {
                return true;
            }
            return base.CanConvertTo(context, destinationType);
        }

        public override object ConvertTo(System.ComponentModel.ITypeDescriptorContext context, System.Globalization.CultureInfo culture, object value, Type destinationType)
        {
            if (value is MyChild c && destinationType == typeof(string))
            {
                return c.Name;
            }

            return base.ConvertTo(context, culture, value, destinationType);
        }
    }
}
```

And when it runs:

![image](https://user-images.githubusercontent.com/8291187/99579944-85bc0800-29a4-11eb-806b-8d0a9f7d2c37.png)


My main concern here is the new checks for `!OperatingSystem.IsBrowser()` could break existing apps/components using TypeDescriptors and TypeConverters.

Reply:
I agree with @eerhardt that I don't see anything obvious on why that API shouldn't be supported on Browser. It might be interesting to look at why the analyzer flagged it as not supported in the first case, maybe there is a corner case throwing PNSE, and we should probably just return null in that case instead if possible.

@marek-safar @steveisok could you help us here?

Reply:
> My main concern here is the new checks for !OperatingSystem.IsBrowser() could break existing apps/components using TypeDescriptors and TypeConverters.

I agree too, if it is unconditionally work for browser we should remove the attribute, and then of course those `!OperatingSystem.IsBrowser()`checks will be removed, we need to make the decision for `TypeDescriptor.CreateInstance()` first

Reply:
this unfortunately looks like a false positive that fell through the cracks.  We can fix it, but I don't think it'll be for 5.0 b/c we don't service the ref assemblies :-(.

Reply:
> we need to make the decision for TypeDescriptor.CreateInstance() first.

I think we should remove the unsupported attribute from this and the core APIs that are causing this to be marked as not supported. 

@buyaa-n I can review TypeConverter in master and see what attributes are not correct, and then we can remove those and adjust this PR based on that.

Reply:
> @buyaa-n I can review TypeConverter in master and see what attributes are not correct, and then we can remove those and adjust this PR based on that.

Great, thank you @safern

Reply:
Ok, so I looked through the APIs that are marked as `UnsupportedOSPlatform("browser")` under `System.ComponentModel.TypeConverter` and to me it seems like all were false positives. I do see a common case where all are using `Activator.CreateInstance` or `Type.GetConstructor` which might have been the common pattern where the analyzer is flagging APIs are unsupported. These are the APIs that are marked as not supported:

https://github.com/dotnet/runtime/blob/aebd598664e7d931eee19eb1779c0ace5d1c3600/src/libraries/System.ComponentModel.TypeConverter/src/System/ComponentModel/LicenseManager.cs#L121
https://github.com/dotnet/runtime/blob/aebd598664e7d931eee19eb1779c0ace5d1c3600/src/libraries/System.ComponentModel.TypeConverter/src/System/ComponentModel/LicenseManager.cs#L132
https://github.com/dotnet/runtime/blob/aebd598664e7d931eee19eb1779c0ace5d1c3600/src/libraries/System.ComponentModel.TypeConverter/src/System/ComponentModel/MaskedTextProvider.cs#L472
https://github.com/dotnet/runtime/blob/aebd598664e7d931eee19eb1779c0ace5d1c3600/src/libraries/System.ComponentModel.TypeConverter/src/System/ComponentModel/TypeDescriptionProvider.cs#L54
https://github.com/dotnet/runtime/blob/aebd598664e7d931eee19eb1779c0ace5d1c3600/src/libraries/System.ComponentModel.TypeConverter/src/System/ComponentModel/TypeDescriptor.cs#L433
https://github.com/dotnet/runtime/blob/aebd598664e7d931eee19eb1779c0ace5d1c3600/src/libraries/System.ComponentModel.TypeConverter/src/System/Security/Authentication/ExtendedProtection/ExtendedProtectionPolicyTypeConverter.cs#L21

@steveisok @eerhardt do you agree that all these should be supported? That is the way it looks to me.

Reply:
It looks like those should all be supported. I'd like to understand why the tool gave a false positive because:

1. Maybe we are missing something by just doing code inspection. And these APIs should really be flagged.
2. Maybe there is a bug in the tool, and we marked a bunch of other APIs as unsupported, when really they should be.

Reply:
> It looks like those should all be supported. I'd like to understand why the tool gave a false positive

Totally agree, @safern do you have the answers to that and/or do want to dig into it? if you have no capacity for this i might dig in later, for now, i am gonna update the conditionals i added for TypeDescriptor.CreateInstance() references with suppress and create a separate issue for tracking and fixing this issue

Reply:
I'm going to let @steveisok or @lewing answer that as they are familiar with the tool that flagged them.

Reply:
> I'm going to let @steveisok or @lewing answer that as they are familiar with the tool that flagged them.

Sure, whatever these APIs are annotated in 5.0, need to be handled separately in case we want the fix ported to 5.0

Reply:
We wouldn't be able to port fixes to 5.0 since we don't update ref assemblies.  Suppressing and filing an issue sounds good here.

Reply:
We do have the ability to service the ref pack this year as work was done to support that, but I think it is a little overkill for this unless we think is critical. cc: @ericstj

Reply:
I always thought the plan of record was to ship the "sidecar" annotation information in the analyzer itself, in the cases where we got the attributes wrong in the ref pack.

Granted - I don't think these issues in ComponentModel are enough to warrant doing that work. But if we were to try to fix the annotations (say something else critical was found), I thought that was the way we were going to do it.

Reply:
> We do have the ability to service the ref pack this year as work was done to support that, but I think it is a little overkill for this unless we think is critical. cc: @ericstj

Ah, OK; thanks.

> I always thought the plan of record was to ship the "sidecar" annotation information in the analyzer itself, in the cases where we got the attributes wrong in the ref pack.

Yeah, that was the plan if we needed to invoke it. Neither servicing approach would be justified unless we have evidence this is causing customer pain though.

Reply:
I don't see any need to backport any of these changes to 5.0

Reply:
As we don't want those updates in 5.0 i have removed unsupported browser from TypeDescriptor.CreateInstance, TypeDescriptionProvider.CreateInstance and related updates. Created [issue ](https://github.com/dotnet/runtime/issues/45162) for handling other APIs (I can update them with this PR too, if that is preferred)

Reply:
As @eerhardt's last comment resolved as he suggested marking this resolved